File: tests/fixtures/valid.swift
import Foundation

struct User {
    let id: Int
    var name: String

    func rename(to newName: String) {
    }
}

extension User {
    func greeting() -> String {
        "Hello"
    }
}

func makeUser() -> User {
    User(id: 1, name: "Ana")
}

File: tests/fixtures/valid.ts
import { readFileSync } from "fs";

interface User {
    id: number;
    name: string;
}

class UserImpl implements User {
    constructor(public id: number, public name: string) {}

    rename(newName: string): void {
        this.name = newName;
    }
}

namespace UserUtils {
    export function greeting(user: User): string {
        return "Hello";
    }
}

function makeUser(): User {
    return new UserImpl(1, "Ana");
}

const greet = (name: string): string => `Hello, ${name}`;
